clappr/clappr-ios
# Replay Button

Goal
---
To create the replay feature, by showing a replay button at the end of the video, which when clicked would start the video from the beginning.

How to Test
---
1. Start the sample
1. Play the video
1. When the video ends, a replay button should show up instead of the video poster
1. Tap that button.
The video should start playing from the beginning.

### Seek & Drawer Scenarios
The media control is now avaible as the video ends, so we've had to define the button's behavior for other interactions as well, such as seeking and showing / hiding drawers.

#### Seek
1. Play a video and when it ends, the media control should show up with the replay button
1. Seek back
The replay button should turn into a play button, since you're not at the end of the video anymore
1. Play the video
1. When the video reaches the end once again, the replay should show up again normally.

#### Drawer
1. Add this chunk code on `ViewController.swift#110`:
```swift
class CustomBottomDrawerPlugin: BottomDrawerPlugin {
    open class override var name: String {
        return "CustomBottomDrawerPlugin"
    }

    override var placeholder: CGFloat {
        return 32.0
    }

    override func render() {
        super.render()

        view.backgroundColor = .red
    }
}
```
2. Add this one to `ViewController.swift`, before the Player is instantiated, around `line 93`:
```swift
Player.register(plugins: [CustomBottomDrawerPlugin.self])
```

3. Run the sample and play the video
3. Open the drawer and let it open until the video ends.
The media control should not be shown.
3. Closing the drawer should show the media control with the replay button.
